commons/src/main/java/com/epam/eco/kafkamanager/PartitionByKeyResolverImpl.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ *  Copyright 2023 EPAM Systems
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License.  You may obtain a copy
+ *  of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ *******************************************************************************/
+package com.epam.eco.kafkamanager;
+
+import org.apache.kafka.common.utils.Utils;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * @author Mikhail_Vershkov
+ */
+
+public class PartitionByKeyResolverImpl implements PartitionByKeyResolver {
```

Comment:
Have you tried to find already implemented alg. inside Apache Kafka library?

Reply:
I get this algorithm from kafka org.apache.kafka.clients.producer.internals.DefaultPartitioner. This is real default kafka partitioner.